doc/upgrading.md

```diff
@@ -0,0 +1,40 @@
+# Managing Upgrades/Downgrades
+This guide describes how to upgrade your SPIRE deployment, as well as the compatibility guarantees that SPIRE users can expect.
+
+## SPIRE Versioning
+SPIRE versions are expressed as **x.y.z**, where **x** is the major version, **y** is the minor version, and **z** is the patch version, following Semantic Versioning terminology. The SPIRE project is currently pre-1.0, so minor SPIRE releases act like major releases and patch releases act like minor releases. Despite the fact that the project is pre-1.0, version compatibility guarantees are already in place.
+
+### SPIRE Server Compatibility
+Version skew within a SPIRE server cluster is supported within +/- 1 minor version. In other words, the newest and oldest SPIRE server instances in any given cluster must be within one minor version of each other.
+
+For example:
+* Newest SPIRE server instance is at 0.9.3
+* Other SPIRE server instances are supported at 0.9.x and 0.8.x
```

Comment:
> Sorry, I'm a bit confused. If 3 is the minor version in 0.9.3, wouldn't +/- one minor version be 0.9.4, 0.9.3, and 0.9.2 and no 0.8.anything?

Related to our above comments, I took the absolute stance, so minor version would be in position 0.x.0. In this context, it would feel weird to say that given version 0.9.3, "9" is the major version (because it is clearly not when prefixed with "0")

> Having these examples is good, but I do prefer the version-skew-policy k8s style where the example versions are listed explicitly like 1.13, 1.12, and 1.11 rather than with "x." Maybe there's a reason you chose that presentation.

I put `x` because I don't know how many releases will fall under them (for instance, today 0.8.4 is the latest but we could cut 0.8.5 in the future, and that version would apply here)

Reply:
Continuing on with my big/little crazy idea:

"Version skew within a SPIRE server cluster is supported within one big version. In other words, the newest and oldest SPIRE server instances in any given cluster must be within one big version of each other.

 For example:
 * Newest SPIRE server instance is at 0.9.3
 * Other SPIRE server instances are supported at 0.9.x and 0.8.x 
"